metaflow/datatools/s3op.py

```diff
@@ -97,14 +109,41 @@ def format_triplet(prefix, url="", local=""):
 def normalize_client_error(err):
     error_code = err.response["Error"]["Code"]
     try:
-        return int(error_code)
+        err_code = int(error_code)
+        # We "normalize" retriable server errors to 503. These are also considered
+        # transient by boto3 (see:
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/guide/retries.html)
+        if err_code in (500, 502, 504):
+            err_code = 503
```

Comment:
rather than swallowing the original error code, can we just retry for 500, 502, 5043 and 504 (in L:184)

Reply:
Are you suggesting I have multiple "values" for transient errors? As of now, in line 184, we return just a ERROR_TRANSIENT_RETRY so the exact code is anyways swallowed there. I could have TRANSIENT_RETRY_500, TRANSIENT_RETRY_502, TRANSIENT_RETRY_504 and TRANSIENT_RETRY_503 if you want but it seems like since the behavior is the same going back out, it won't matter much.

Reply:
No - in L:190 you could simply do 
```
elif error_code in (500, 502, 503, 504):
                to_return = {"error": ERROR_TRANSIENT, "raise_error": err}

```

Reply:
Sure, I can make that change and return 503 for all the text and then catch all these other ones explicitly.